### PR TITLE
Rename all mentions in docs of `DataSet` to `Dataset`

### PR DESCRIPTION
**Pull request by merelcht**
## Description
Part of https://github.com/kedro-org/kedro/issues/2129

## Development notes
Non docs changes done in https://github.com/kedro-org/kedro/pull/3147/files

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -112,16 +112,16 @@
...
         "kedro.io.core.Version",
         "kedro.io.data_catalog.DataCatalog",
-        "kedro.io.memory_dataset.MemoryDataSet",
-        "kedro.io.partitioned_dataset.PartitionedDataSet",
+        "kedro.io.memory_dataset.MemoryDataset",
+        "kedro.io.partitioned_dataset.PartitionedDataset",
         "kedro.pipeline.pipeline.Pipeline",
         "kedro.runner.runner.AbstractRunner",
-        "kedro.runner.parallel_runner._SharedMemoryDataSet",
+        "kedro.runner.parallel_runner._SharedMemoryDataset",
```
Comment (deepyaman) [inline docs/source/conf.py:124]:
```suggestion
```

File: docs/source/configuration/advanced_configuration.md [hunk 2]
```diff
@@ -60,8 +60,8 @@ bucket_name: "my_s3_bucket"
 key_prefix: "my/key/prefix/"
 
 datasets:
-    csv: "pandas.CSVDataSet"
-    spark: "spark.SparkDataSet"
+    csv: "pandas.CSVDataset"
```
Comment (deepyaman) [inline docs/source/configuration/advanced_configuration.md:63]: These dataset names won't work unless the user upgrades `kedro-datasets`; do we need some sort of communication around this? Does it need to be in the docs?

**Review by deepyaman — APPROVED** [review]
One minor suggestion, but otherwise LGTM

File: docs/source/configuration/advanced_configuration.md (hunk 2 above)
Comment (merelcht) [reply docs/source/configuration/advanced_configuration.md:63]: This is going into `develop`, I think we should add some messaging in the `main` branch soon so users already know this is happening before `0.19.0` and `2.0.0` for datasets.

**Review by stichbury — APPROVED** [review]
LGTM 💯

**Pull request merged**